### PR TITLE
Fix missing nudge context #2274

### PR DESCRIPTION
**Pull request by shshshl09**

### CODE REVIEW

**Review by tommy5oo — APPROVED** [review]

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt [hunk 1]
```diff
@@ -135,34 +142,43 @@ class NudgeAgent(
...
 
-    internal fun clearNudgeData() {
+    internal fun clearNudgeData(dialogRequestId: String) {
         executor.submit {
-            Logger.d(TAG, "clearNudgeData")
-            nudgeData = null
+            Logger.d(TAG, "clearNudgeData target dialogRequestId:$dialogRequestId, nudgeData: $nudgeData")
+            if (nudgeData?.dialogRequestId == dialogRequestId) {
+                nudgeData = null
+                currentNudgeRelatedDirectiveHandlingListener = null
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt:150]: at this point, 
if clearNudgeData() called from playSynchronizer, It seems currentNudgeRelatedDirectiveHandlingListener  is not removed from directiveSequencer and directiveGroupProcessor. 
So how about call them explicitly.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt [hunk 2]
```diff
@@ -90,9 +90,16 @@ class NudgeAgent(
...
+                    if(!receivedPreparedOrStartedPlaySyncObject) {
+                        if(preparedObjectExist || startedObjectExist) {
+                            receivedPreparedOrStartedPlaySyncObject = true
+                        }
+                    }
+
                     log += "preparedObject Exist: $preparedObjectExist, startedObject Exist: $startedObjectExist"
-                    if (!preparedObjectExist && !startedObjectExist) {
-                        clearNudgeData()
+                    if (!preparedObjectExist && !startedObjectExist && receivedPreparedOrStartedPlaySyncObject) {
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt:101]: I this it should be !receivedPreparedOrStartedPlaySyncObject.
isn't it?

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt:101]: yes. it means received before.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt (hunk 1 above)
Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/nudge/NudgeAgent.kt:150]: The currentNudgeRelatedDirectiveHandlingListener is removed from directiveSequencer and directiveGroupProcessor when a releated directives finished. This has nothing to do with playSynchronizer. 
Look at [DirectiveGroupHandlingListener] class implementation.

**Review by Yuseon-Han — APPROVED** [review]

**Pull request merged**